element_calcium_imaging/imaging_preprocess.py

```diff
@@ -0,0 +1,1283 @@
+        :param scan_schema_name: schema name on the database server to activate the `scan` module
+         - may be omitted if the `scan` module is already activated
+        :param create_schema: when True (default), create schema in the database if it does not yet exist.
+        :param create_tables: when True (default), create tables in the database if they do not yet exist.
+        :param linking_module: a module name or a module containing the
+         required dependencies to activate the `imaging` module:
+         + all that are required by the `scan` module
+    """
+
+    if isinstance(linking_module, str):
+        linking_module = importlib.import_module(linking_module)
+    assert inspect.ismodule(
+        linking_module
+    ), "The argument 'dependency' must be a module's name or a module"
+
+    global _linking_module
+    _linking_module = linking_module
+
+    scan.activate(
+        scan_schema_name,
+        create_schema=create_schema,
+        create_tables=create_tables,
+        linking_module=linking_module,
+    )
+    schema.activate(
+        imaging_schema_name,
+        create_schema=create_schema,
+        create_tables=create_tables,
+        add_objects=_linking_module.__dict__,
+    )
+
+
+# -------------- Table declarations --------------
+
+
+@schema
+class PreProcessMethod(dj.Lookup):
+    definition = """  #  Method/package used for pre-processing
+    preprocess_method: varchar(16)
+    ---
+    preprocess_method_desc: varchar(1000)
+    """
+
+
+@schema
+class PreProcessParamSet(dj.Lookup):
+    definition = """  #  Parameter set used for pre-processing of calcium imaging data
+    paramset_idx:  smallint
+    ---
+    -> PreProcessMethod
+    paramset_desc: varchar(128)
+    param_set_hash: uuid
+    unique index (param_set_hash)
+    params: longblob  # dictionary of all applicable parameters
+    """
+
+    @classmethod
+    def insert_new_params(
+        cls, preprocess_method: str, paramset_idx: int, paramset_desc: str, params: dict
+    ):
+        param_dict = {
+            "preprocess_method": preprocess_method,
+            "paramset_idx": paramset_idx,
+            "paramset_desc": paramset_desc,
+            "params": params,
+            "param_set_hash": dict_to_uuid(params),
+        }
+        q_param = cls & {"param_set_hash": param_dict["param_set_hash"]}
+
+        if q_param:  # If the specified param-set already exists
+            pname = q_param.fetch1("paramset_idx")
+            if pname == paramset_idx:  # If the existed set has the same name: job done
+                return
+            else:  # If not same name: human error, trying to add the same paramset with different name
+                raise dj.DataJointError(
+                    "The specified param-set already exists - name: {}".format(pname)
+                )
+        else:
+            cls.insert1(param_dict)
+
+
+@schema
+class PreProcessParamSteps(dj.Manual):
```

Comment:
I think we should think of `preprocess` as one word and not capitalize the second `p`
```suggestion
class PreprocessParamSteps(dj.Manual):
```

Reply:
Thanks.  Updated.